### ISSUE TITLE
[Feature] Add lables for storage classes and chunks distribution rules based on them

### ISSUE THREAD

**Issue by Onlinehead** [ytsaurus/ytsaurus#15]
YT has an a "media" storage class abstraction, which allow to set a storage class in Cypress.
The proposed feature is about extending the storage mechanism to support some advantage chunk distribution scenarios.

**Idea**
1. Add an additional abstraction "label" which can be associated with each "media" on the node level and represents more details about it, like redundancy or performance.
2. Add a possibility to set policy for chunk distribution based on labels.
For example: we have 3 nodes for blobs, each of it has a single "ssd_blobs" media. Also, we have a table `/my/mytable`, with replication_factor=3.
On the first node we can set label "type=redundant", on other 2 - "type=ephemeral". 
After that, we can set a distribution policy for the table `/my/mytable`, which will allow us to make sure we always have one replica on the "type=redundant" media and all others on the "type=ephemeral". 
3. Optional: make priority during data read on the labels level.

**Motivation**
In clouds you can have an expensive, but redundant and relatively slow network volumes and fast and cheap, but not redundant local SSDs. Enabling chunk distribution between them can speed up operations and at the same time reduce the cost of ownership, because you can make sure you have at least 1 copy on the attached storage (which has an underlayer replication) even if all nodes will down and deleted, but at the same time you can store a major part of the data on local SSDs which are fast and relatively cheap for day-to-day operations - until everything is fine we can use 2 replicas on local SSDs (ephemeral) even to replace or restore a redundant storage on a network disk.

**Comment by gritukan** [comment]
Hi!

I am not sure if I understood correctly items (1) and (2). Is it true that there are only two possible values for "type" flag: "redundant" for media for which storing data with replication factor 1 is reliable since they correspond to some external storages with replication (e.g. S3 or NBD) and "ephemeral" for regular disks?

Replicas prioritization based on media properties was discussed internally many times and probably we will revisit this idea.

**Comment by Onlinehead** [comment]
@gritukan 
> Is it true that there are only two possible values for "type" flag

Not really. `type=redundant` is just a key-value, we can use any "groups", like `redundant`, or `cheap`, or `failover_zone` etc.

> some external storages with replication (e.g. S3 or NBD) and "ephemeral" for regular disks?

In a cloud any regular disk you mount are redundant, it is not mandatory an S3 or NBD. Even a regular root volume which any node use to boot is redundant, e.g. for Amazon [EBS](https://aws.amazon.com/ebs/features/):
> Amazon EBS allows you to create storage volumes and attach them to Amazon EC2 instances. Once attached, you can create a file system on top of these volumes, run a database, or use them in any other way you would use block storage. Amazon EBS volumes are placed in a specific Availability Zone where they are automatically replicated to protect you from the failure of a single component. All EBS volume types offer durable snapshot capabilities and are designed for 99.999% availability.

At the same time a cloud has an "ephemeral" disks which CAN failure by design and you can also loose data in some cases like stop&start an instance. For example - that is an [Instance Storage](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html) from AWS or a [Local SSD](https://cloud.google.com/compute/docs/disks/local-ssd) from Google.
So the idea is to make it able (in an AWS terminology) to put one copy to the EBS, and others - to the instance storage.
